### PR TITLE
feat(client): Embed config values in the HTML.

### LINKED ISSUE #2585: Discussion: remove /config endpoint, instead have the options set during the build step or via the server
Currently we call the /config endpoint on page load. 
The is the data we load:

```
res.json({
      // The `__cookies_check` cookie is set in client code
      // to see if cookies are enabled. If cookies are disabled,
      // the `__cookie_check` cookie will not arrive.
      allowedParentOrigins: allowedParentOrigins,
      authServerUrl: authServerUrl,
      cookiesEnabled: !!req.cookies['__cookie_check'],
      marketingEmailServerUrl: marketingEmailApiServerUrl,
      marketingEmailPreferencesUrl: marketingEmailPreferencesUrl,
      oAuthClientId: clientId,
      oAuthUrl: oauthServerUrl,
      // req.lang is set by abide in a previous middleware.
      language: req.lang,
      metricsSampleRate: metricsSampleRate,
      profileUrl: profileServerUrl
    });
```

Instead can we write this data into the template using the server?
(If index.html has to be a static resource then the build process can write the required data into the template, see `view-source:https://oauth-stable.dev.lcip.org/console/login` ) 
i.e 
```
<meta name="fxa-oauth-console/config/environment" content="%7B%22modulePrefix%22%3A%22fxa-oauth-console%22%2C%22simple-auth%22%3A%7B%22authorizer%22%3A%22authorizer%3Acustom%22%7D%2C%22servers%22%3A%7B%22oauth%22%3A%22https%3A//oauth-stable.dev.lcip.org%22%2C%22oauthInternal%22%3A%22https%3A//oauth-stable.dev.lcip.org%22%2C%22oauthUriParsed%22%3A%7B%22protocol%22%3A%22https%3A%22%2C%22slashes%22%3Atrue%2C%22auth%22%3Anull%2C%22host%22%3A%22oauth-stable.dev.lcip.org%22%2C%22port%22%3Anull%2C%22hostname%22%3A%22oauth-stable.dev.lcip.org%22%2C%22hash%22%3Anull%2C%22search%22%3Anull%2C%22query%22%3Anull%2C%22pathname%22%3A%22/v1%22%2C%22path%22%3A%22/v1%22%2C%22href%22%3A%22https%3A//oauth-stable.dev.lcip.org/v1%22%7D%2C%22oauthInternalUriParsed%22%3A%7B%22protocol%22%3A%22https%3A%22%2C%22slashes%22%3Atrue%2C%22auth%22%3Anull%2C%22host%22%3A%22oauth-stable.dev.lcip.org%22%2C%22port%22%3Anull%2C%22hostname%22%3A%22oauth-stable.dev.lcip.org%22%2C%22hash%22%3Anull%2C%22search%22%3Anull%2C%22query%22%3Anull%2C%22pathname%22%3A%22/v1%22%2C%22path%22%3A%22/v1%22%2C%22href%22%3A%22https%3A//oauth-stable.dev.lcip.org/v1%22%7D%2C%22profileUriParsed%22%3A%7B%22protocol%22%3A%22https%3A%22%2C%22slashes%22%3Atrue%2C%22auth%22%3Anull%2C%22host%22%3A%22stable.dev.lcip.org%22%2C%22port%22%3Anull%2C%22hostname%22%3A%22stable.dev.lcip.org%22%2C%22hash%22%3Anull%2C%22search%22%3Anull%2C%22query%22%3Anull%2C%22pathname%22%3A%22/profile/v1%22%2C%22path%22%3A%22/profile/v1%22%2C%22href%22%3A%22https%3A//stable.dev.lcip.org/profile/v1%22%7D%7D%2C%22environment%22%3A%22production%22%2C%22baseURL%22%3A%22/console/%22%2C%22locationType%22%3A%22auto%22%2C%22EmberENV%22%3A%7B%22FEATURES%22%3A%7B%7D%7D%2C%22APP%22%3A%7B%7D%2C%22contentSecurityPolicyHeader%22%3A%22Content-Security-Policy-Report-Only%22%2C%22contentSecurityPolicy%22%3A%7B%22default-src%22%3A%22%27none%27%22%2C%22script-src%22%3A%22%27self%27%22%2C%22font-src%22%3A%22%27self%27%22%2C%22connect-src%22%3A%22%27self%27%22%2C%22img-src%22%3A%22%27self%27%22%2C%22style-src%22%3A%22%27self%27%22%2C%22media-src%22%3A%22%27self%27%22%7D%2C%22exportApplicationGlobal%22%3Afalse%7D" />
```

@shane-tomlinson @zaach thoughts? 

The sensitive config items are lang and cookie check.
This might fix issues such as https://github.com/mozilla/fxa-content-server/issues/2557 and speed up load time.

### LINKED ISSUE #4141: Sentry Error: System unavailable, try again soon
There are between 5500 and 7700 of these ~~an hour~~ a day reported to Sentry.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
This serves multiple purposes:
* Eliminate errors when calling /config (we have a huge number of errors)
* Reduce the number of XHR requests needed on startup and speed load time.

fixes #2585
fixes #4141

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@vladikoff - this is for you.

File: server/lib/routes/get-config.js [hunk 1]
```diff
@@ -2,52 +2,19 @@
...
-var oauthServerUrl = config.get('oauth_url');
-var profileServerUrl = config.get('profile_url');
-var allowedParentOrigins = config.get('allowed_parent_origins');
-var marketingEmailApiServerUrl = config.get('marketing_email.api_url');
-var marketingEmailPreferencesUrl = config.get('marketing_email.preferences_url');
-
 module.exports = function () {
   var route = {};
 
+  // support for this route was removed in PR #4147, config
```
Comment (shane-tomlinson) [inline server/lib/routes/get-config.js]: @vladikoff, @jrgm - think we should leave `/config` with all of this values around for a train to help with the crossover period after train cut? There is a tiny window of failure:

1. Old JS that requests `/config` is downloaded, request to `/config has not yet occurred.
2. content-server swap happens.
3. `/config` is requested. Oh oh, 410.

Comment (vladikoff) [inline server/lib/routes/get-config.js]: > think we should leave /config with all of this values around for a train to help with the crossover period after train cut? 

👍

File: server/lib/routes/get-config.js [hunk 2]
```diff
@@ -2,52 +2,19 @@
...
-      env: env,
-      // req.lang is set by abide in a previous middleware.
-      language: req.lang,
-      marketingEmailPreferencesUrl: marketingEmailPreferencesUrl,
-      marketingEmailServerUrl: marketingEmailApiServerUrl,
-      oAuthClientId: clientId,
-      oAuthUrl: oauthServerUrl,
-      profileUrl: profileServerUrl
-    });
+    res.status(410).send('gone');
```
Comment (vladikoff) [inline server/lib/routes/get-config.js]:
```
FAIL: front end routes - #https get http://127.0.0.1:3030/config (7ms)
AssertionError: expected 410 to equal 200
```

We cannot remove this until we are 1 train ahead.

File: app/scripts/lib/config-loader.js [hunk 3]
```diff
@@ -25,61 +22,76 @@ define(function (require, exports, module) {
...
-      return self._configPromise;
-    },
+      return p(config);
+    }
   };
 
-  var t = function (msg) {
-    return msg;
-  };
+  var t = msg => msg;
```
Comment (vladikoff) [inline app/scripts/lib/config-loader.js]: cool!

File: server/templates/pages/src/index.html [hunk 4]
```diff
@@ -33,6 +33,8 @@
           <script src="/bower_components/html5shiv/dist/html5shiv.js"></script>
           <!-- endbuild -->
         <![endif]-->
+
+        <meta name="fxa-content-server/config" content="{{ config }}" />
```
Comment (vladikoff) [inline server/templates/pages/src/index.html]: should we put this with the rest of the `meta` elements before other css / js stuff?

Comment (shane-tomlinson) [inline server/templates/pages/src/index.html]: 👍 making the update now.

File: server/lib/routes/get-config.js (hunk 2 above)
Comment (shane-tomlinson) [inline server/lib/routes/get-config.js]: On it.

**Comment by shane-tomlinson** [comment]
@vladikoff - made the requested changes.

File: server/lib/routes/get-config.js (hunk 1 above)
Comment (shane-tomlinson) [inline server/lib/routes/get-config.js]: How in the hell do I say I've done it?

File: server/templates/pages/src/index.html [hunk 5]
```diff
@@ -9,6 +9,7 @@
         <meta name="viewport" content="width=device-width">
         <meta name="referrer" content="origin">
         <meta name="robots" content="noindex,nofollow">
+        <meta name="fxa-content-server/config" content="{{ config }}" />
```
Comment (vladikoff) [inline server/templates/pages/src/index.html]: when I run this with production config (against circle), I get ` <meta name=fxa-content-server/config content=""> `

example: build and run `CONFIG_FILES=server/config/local.json,server/config/production.json,tests/ci/config_circleci.json node_modules/.bin/grunt serverproc:dist`

output:

```
<!DOCTYPE html><!--[if lt IE 10]>      <html class="lt-ie10" dir="ltr" lang="en"> <![endif]--><!--[if gte IE 10]><!--> <html dir=ltr lang=en><!--<![endif]--> <head> <meta charset=utf-8> <meta http-equiv=X-UA-Compatible content="IE=edge"> <title>Firefox Accounts</title> <meta name=description content=""> <meta name=viewport content="width=device-width"> <meta name=referrer content=origin> <meta name=robots content=noindex,nofollow> <meta name=fxa-content-server/config content=""> <link rel=stylesheet href="http://127.0.0.1:3030/styles/9669d56c.en.css" integrity="sha512-wt5yv7xY8IyZN7K7OAhCH4MLHoSZKM8WNSyxFWySILCNL99fqWexCN8KNZbukigmh4QNxLN6CtK8j4ZpLsDqww==" crossorigin=anonymous> <script src="http://127.0.0.1:3030/scripts/3f80b962.head.js" integrity="sha512-znK0qshAgi1SasO6Q4bLnAZiEXXDCyHCKseNwYnDCgZpTi4XpqB6oxJuwwCpzRHRiZNc3ijWiR5wKPLY0GHyXg==" crossorigin=anonymous></script><!--[if lt IE 9]>
```

Comment (shane-tomlinson) [inline server/templates/pages/src/index.html]: Ah yes... I have to tunnel that variable through the build process. Great catch.

Comment (shane-tomlinson) [inline server/templates/pages/src/index.html]: Fixed.

**Comment by vladikoff** [comment]
woohoo!

**Pull request merged**